### PR TITLE
In ServiceBus receiver set max_message_count to 1 for receive_messages

### PR DESCRIPTION
**Pull request by stefpiatek**
Hopefully haven't misunderstood what you were looking for and tracked down the relevant changes required. Also is there a specific branch I should be merging into?

I ended up changing some logic which didn't make sense if the default value was never None. Not sure if it's worth putting a guard clause in to make sure that the `max_message_count` is greater than 0?

Happy to make changes, so let me know 😄 

Closes Azure/azure-sdk-for-python#14709

### CODE REVIEW

**Comment by stefpiatek** [comment]
Oh also I'm making a hillarious last minute attempt for hacktober-fest, would you be happy to tag the PR with the `hacktoberfest-accepted` label so it counts once you're happy with it an it's approved?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py:233]: I think we could keep the old logic allowing `max_message_count = None` which has the special meaning that it's fully depending on the "prefetch" config on the receiver to decide on the receiving message count. For users they don't care/know the exact amount they want to receive but just want to consume the message as fast as possible. They could do:
```python
receiver = sbclient.get_quque_receiver(..., prefetch=1000)
with receiver:
    while more_message:
        messages = receiver.receive(max_message_count=None)
        # handle each message
        more_message = len(messages) > 0
```
Does this make sense to you?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py:402]: You've brought great point on checking max_message_count > 0 if given, that's a valid scenario and we should probably do that like:

```python
If max_message_count is not None and max_message_count <= 0:
    raise ValueError("The max_message_count must be greater than 0").
```

Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py:402]: `async receive_messages` should apply the change as well which is here:
https://github.com/Azure/azure-sdk-for-python/blob/7a730caaeb6ca118a95b86df08448b2aac9713ae/sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py#L396

**Comment by yunhaoling** [comment]
appreciate the PR @stefpiatek! left some comments and feel free to share your thoughts!

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py
Comment (stefpiatek) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py:233]: ah yes that sounds like a good thing to have there, perfect, I'll revert it

**Comment by stefpiatek** [comment]
No worries, eventually found some time to address them - let me know if there's anything else I can do for this

**Comment by yunhaoling** [comment]
@stefpiatek I did some minor changes like changelog/test/mypy update, we shall be good to go after tests pass :)

**Review by yunhaoling — APPROVED** [review]
thanks @stefpiatek for your contribution! glad to be collaborating with you on this :)
I'm gonna merge it for you.

**Comment by stefpiatek** [comment]
Likewise! Also it looks like this isn't coming up for hacktoberfest, maybe it's just because it's approved after October potentially? Would you mind adding the `hacktoberfest-accepted` tag to see if that does the trick?

**Pull request merged**
